components/ActionSheet/src/MDCActionSheetController.m

```diff
@@ -0,0 +1,331 @@
+
+  if (initialLayout == false) {
+    [self firstLayout];
+  }
+}
+
+-(void)firstLayout {
+  [self.header setNeedsLayout];
+  [self.header layoutIfNeeded];
+
+  CGFloat width = CGRectGetWidth(self.view.bounds);
+  CGFloat height = CGRectGetHeight(self.header.frame) + [_tableView tableHeightForWidth:width];
+  CGRect tableFrame = _tableView.tableView.frame;
+  tableFrame.origin.y = CGRectGetHeight(self.header.frame);
+  _tableView.tableView.frame = tableFrame;
+  self.preferredContentSize = CGSizeMake(width, height);
+  initialLayout = true;
+}
+
+- (void)viewWillAppear:(BOOL)animated {
+  [super viewWillAppear:animated];
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+  self.mdc_bottomSheetPresentationController.delegate = self;
+#pragma clang diagnostic pop
+
+  self.mdc_bottomSheetPresentationController.dismissOnBackgroundTap =
+      _transitionController.dismissOnBackgroundTap;
+  [self.view layoutIfNeeded];
+}
+
+- (BOOL)accessibilityPerformEscape {
+  if (!self.dismissOnBackgroundTap) {
+    return NO;
+  }
+  [self dismissViewControllerAnimated:YES completion:nil];
+  return YES;
+}
+
+- (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentContainer>)container {
+  [super preferredContentSizeDidChangeForChildContentContainer:container];
+  [self.presentationController
+      preferredContentSizeDidChangeForChildContentContainer:self];
+}
+
+- (UIScrollView *)trackingScrollView {
+  return _transitionController.trackingScrollView;
+}
+
+- (void)setTrackingScrollView:(UIScrollView *)trackingScrollView {
+  _transitionController.trackingScrollView = trackingScrollView;
+}
+
+- (BOOL)dismissOnBackgroundTap {
+  return _transitionController.dismissOnBackgroundTap;
+}
+
+-(void)setDismissOnBackgroundTap:(BOOL)dismissOnBackgroundTap {
+  _transitionController.dismissOnBackgroundTap = dismissOnBackgroundTap;
+  self.mdc_bottomSheetPresentationController.dismissOnBackgroundTap = dismissOnBackgroundTap;
+}
+
+- (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:
+    (id<UIViewControllerTransitionCoordinator>)coordinator {
+  [super viewWillTransitionToSize:size withTransitionCoordinator:coordinator];
+  CGRect frame = self.view.frame;
+  frame.size = size;
+  self.view.frame = frame;
+  CGRect headerFrame = self.header.frame;
+  headerFrame.size.width = size.width;
+  self.header.frame = headerFrame;
+  [self.header setNeedsLayout];
+  [self.header layoutIfNeeded];
+
+  CGFloat height = CGRectGetHeight(self.header.frame) + [_tableView tableHeightForWidth:size.width];
+  CGSize updatedSize = CGSizeMake(size.width, height);
+  self.preferredContentSize = updatedSize;
+}
+
+-(void)setTransitioningDelegate:(id<UIViewControllerTransitioningDelegate>)transitioningDelegate {
+  NSAssert(NO, @"MDCActionSheetController.transitionDelegate cannot be changed");
+  return;
+}
+
+-(void)setModalPresentationStyle:(__unused UIModalPresentationStyle)modalPresentationStyle {
```

Comment:
[minornit] Spaces after the `-` and `+` throughout all of your .m files.

Reply:
Done